src/plugins/nvme/nvme-fabrics.c

```diff
@@ -766,3 +767,135 @@ gchar ** bd_nvme_find_ctrls_for_ns (const gchar *ns_sysfs_path, const gchar *sub
     g_ptr_array_add (ptr_array, NULL);  /* trailing NULL element */
     return (gchar **) g_ptr_array_free (ptr_array, FALSE);
 }
+
+
+/**
+ * bd_nvme_get_host_nqn:
+ * @error: (out) (nullable): Place to store error (if any).
+ *
+ * Reads the Host NQN (NVM Qualified Name) value from the `/etc/nvme/hostnqn` file
+ * or generates new one from system DMI data.
```

Comment:
This feels kinda unintuitive, especially because `bd_nvme_get_host_id` doesn't generate new ID when the file doesn't exist, it might be better to have two functions -- one to read nqn from the file and one to generate id.

Reply:
Separated, after mutual agreement.